pkg/groupsync/groups.go

```
@@ -0,0 +1,187 @@
+	// Otherwise, it returns an error.
+	User() (*User, error)
+}
+
+// UserMember represents a user membership of a group.
+type UserMember struct {
+	Usr *User
+}
+
+// IsUser returns whether this Member is a User. Always returns true.
+func (u *UserMember) IsUser() bool {
+	return true
+}
+
+// IsGroup returns whether this Member is a Group. Always returns false.
+func (u *UserMember) IsGroup() bool {
+	return false
+}
+
+// Group returns an error.
+func (u *UserMember) Group() (*Group, error) {
+	return nil, fmt.Errorf("user is not a group")
+}
+
+// User returns the underlying user if this Member.
+func (u *UserMember) User() (*User, error) {
+	return u.Usr, nil
+}
+
+// GroupMember represents a group membership of a group.
+type GroupMember struct {
+	Grp *Group
+}
+
+// IsGroup returns whether this Member is a Group. Always returns true.
+func (g *GroupMember) IsGroup() bool {
+	return true
+}
+
+// IsUser returns whether this Member is a User. Always returns false.
+func (g *GroupMember) IsUser() bool {
+	return false
+}
+
+// Group returns the underlying group of this Member.
+func (g *GroupMember) Group() (*Group, error) {
+	return g.Grp, nil
+}
+
+// User returns an error.
+func (g *GroupMember) User() (*User, error) {
+	return nil, fmt.Errorf("group is not a user")
+}
+
+// Descendants retrieve all users (children, recursively) of the given
+// group ID using the given memberFunc. This function serves mostly as
+// a utility function when implementing ReadGroupClients for when there
+// is no special logic for fetching descendants.
+func Descendants(ctx context.Context, groupID string, memberFunc func(context.Context, string) ([]Member, error)) ([]*User, error) {
+	members, err := memberFunc(ctx, groupID)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching group members: %s, %w", groupID, err)
+	}
+	var users []*User
+	var merr error
+	for _, member := range members {
+		if member.IsUser() {
+			user, _ := member.User()
```

Lint didn't give you problem about not checking err?
And why can we safely ignore the error here?

---

We can ignore the error as the check `member.IsUser()` was true in order to enter the `if` block. so we will never get an error as the member will always be a user at that point.

---

And yeah the linter didn't complain about ignoring the err. It might be smart enough to realize the error will never occur.